x/msg_authorization/internal/keeper/keeper.go

```diff
@@ -0,0 +1,41 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/msg_authorization/internal/types"
+	"time"
+)
+
+type Keeper struct {
+	storeKey sdk.StoreKey
+	cdc      *codec.Codec
+	router   sdk.Router
+}
+
+// NewKeeper constructs a message authorisation Keeper
+func NewKeeper(storeKey sdk.StoreKey, cdc *codec.Codec, router sdk.Router) Keeper {
+	return Keeper{
+		storeKey: storeKey,
+		cdc:      cdc,
+		router:   router,
+	}
+}
+
+func (k Keeper) DispatchActions(ctx sdk.Context, grantee sdk.AccAddress, msgs []sdk.Msg) sdk.Result {
```

Comment:
Wouldn't it be simpler to allow one-delegation for one `sdk.Msg`?
We are already wrapping `dispatchMsg{desiredMsg}` for the `sdk.Msg`. And the sdk allows us to provide multiple messages, so accepting multiple doesn't add any functionality. I would simplify it here.

Reply:
Any concerns with this @aaronc, in the ADR it was for multiple messages so I used it accordingly

Reply:
I think we could do it either way. I don't think it's too complex implementation-wise either way, but if you think it's an important simplification we can do that.

Reply:
I sw the adr defined it that was. You can keep it.

I figured the handler could make multiple calls to the keeper, but really it doesn't matter where the loop is. Leave it or move the array one layer up to the hanlder logic, either is fine with me